### PR TITLE
add functional group norm and dtype device param

### PR DESCRIPTION
**Pull request by marigoold**

### CODE REVIEW

File: python/oneflow/nn/modules/normalization.py [hunk 1]
```diff
@@ -106,6 +150,9 @@ def reset_parameters(self) -> None:
             flow.nn.init.zeros_(self.bias)
 
     def forward(self, input: Tensor) -> Tensor:
+        return group_norm(
+            input, self.num_groups, self.weight, self.bias, self.eps, self.num_channels
+        )
```
Comment (levi131) [inline python/oneflow/nn/modules/normalization.py:155]: 现在直接return一个group_norm接口，和下边之前的实现关系是？

Comment (marigoold) [reply python/oneflow/nn/modules/normalization.py:155]: 增加了一个 nn.functional.group_norm，把之前的实现挪到 group_norm 了，nn.Module 这里就调用 functional 的实现

File: python/oneflow/nn/modules/normalization.py [hunk 2]
```diff
@@ -24,6 +24,43 @@
...
+    See :class:`~oneflow.nn.GroupNorm` for details.
+    """
+    assert len(input.shape) >= 3, "The dimensions of input tensor must larger than 2"
+    if num_channels is None:
+        num_channels = input.shape[1]
+    assert (
+        input.shape[1] == num_channels
+    ), "The channels of input tensor must equal num_channels"
+
+    affine = weight is not None and bias is not None
```
Comment (marigoold) [inline python/oneflow/nn/modules/normalization.py:46]: 这里是相比于原来实现增加的一部分

**Review by levi131 — APPROVED** [review]
LGTM

**Review by wyg1997 — APPROVED** [review]
